docs/blog/2019-03-01-localization-with-gatsby-and-sanity/index.md

```diff
@@ -0,0 +1,272 @@
+
+```js:title=/src/util/index.js
+export const createLocaleTextGetter = languageCode => {
+  const languages = [languageCode, "en"] // last language in array is default;
+  const localize = value => {
+    if (Array.isArray(value)) {
+      return value.map(v => localize(v, languages))
+    } else if (typeof value == "object") {
+      if (/^locale[A-Z]/.test(value._type)) {
+        const language = languages.find(lang => value[lang])
+        return value[language]
+      }
+
+      return Object.keys(value).reduce((result, key) => {
+        result[key] = localize(value[key], languages)
+        return result
+      }, {})
+    }
+    return value
+  }
+
+  return localize
+}
+```
+
+You could use render props for this but I decided on a higher-order component (HOC).
+
+```JSX:title=/src/components/localize.jsx
+import React from "react";
+import Proptypes from "prop-types";
+import { createLocaleTextGetter } from '../../util'; // Or wherever you stashed it
+
+function localize(Component) {
+  return class Localize extends React.Component {
+    static propTypes = {
+      data: Proptypes.object,
+      pageContext: Proptypes.shape({
+        locale: Proptypes.string,
+      }),
+    };
+    constructor(props) {
+      super(props);
+
+      this.getLocalizedContent = createLocaleTextGetter(
+        this.props.pageContext.locale
+      );
+    }
+
+    render() {
+      return (
+        <Component
+          {...this.props}
+          data={this.getLocalizedContent(this.props.data)}
+        />
+      );
+    }
+  };
+}
+
+export default localize;
+```
+
+## Wrap Page Components With the Localize HOC
+
+For each page or template that needs to be localized add your HOC.
+
+```jsx:title=/pages/index.js
+import React from "react"
+import Layout from "../components/Layout"
+import localize from "../components/localize"
+
+const IndexPage = ({ data }) => (
+  <Layout>
+    <h1>{data.sanitySomeDataType.title}</h1>
+  </Layout>
+)
+
+export default localize(IndexPage)
+
+export const query = graphql`
+  query HomeQuery {
+    sanitySomeDataType {
+      title {
+        _type
+        en
+        es
+      }
+    }
+  }
+`
+```
+
+## Generating Pages with the Correct Context
+
+In the `gatsby-node.js` file you need to add a function that generates pages for each supported language with a path prefix for each, such as 'es' for Spanish, and 'fr' for French.
+
+```js:title=gatsby-node.js
+// Get your list of languages from somewhere, env file, config.json, etc
+// for sake of this snippet I am putting it here
+
+const extraLanguages = ["es"]; // English is currently the default so it isn't needed here.
+
+exports.onCreatePage = ({ page, actions }) => {
+  const { createPage, deletePage } = actions;
+
+  deletePage(page);
+  const {context, ...rest}
```

Comment:
Think we're missing an assignment here?

Reply:
Whoops... missed that in an edit. Pushing a fix now.